plugin/core/promise.py

```diff
@@ -184,3 +186,40 @@ def _is_resolved(self) -> bool:
     def _get_value(self) -> Any:
         with self.mutex:
             return self.value
+
+
+opening_files = {}  # type: Dict[str, Tuple[Promise, Callable[[sublime.View], None]]]
+
+
+def open_file(
```

Comment:
I find the placement of this utility function rather weird. `promise.py` is a file containing an implementation of the Promise concept itself. This function doesn't have anything to do with the implementation of promise but is just a utility function that happens to use promise for execution (and would likely use asyncio instead if that would be available).

Reply:
Where would you suggest we place this function?

Reply:
I'm afraid of suggesting "utils" since it seems like you were avoiding that on purpose for some time now. ;)

Would `core/windows.py` work for now, since this is for opening file in a window so somewhat related.